doc/source/cuds_definition.rst

```diff
@@ -0,0 +1,89 @@
+CUDS Data Structures
+====================
+
+The Common Unified Data Structures (CUDS) define the expected
+interface for the modelling engine wrappers, the visualisation api the
+top level CUDS containers (i.e. Mesh, Lattice and Particles) and the
+low level CUDS components (e.g Point, Bond, LatticeNode)
+
+.. note::
+
+   The CUDS API is constantly evolving and changes through the processes
+   of the SSB and the designated workgroups.
+
+
+Modelling engine
+----------------
+
+.. figure:: ./images/engine.png
+
+  **Figure. 1:** UML description of a CUDS modelling engine.
+
+The CUDS engine supports operations to add, get and remove CUDS
+containers from the internal memory of the engine wrapper to setup
+the SD state of the simulations.
+
+When a CUDS container is added the CUDS modelling engine will copy
+into internal memory only the ``uid`` mapping and the CUBA information
+that it can support. The user needs to refer to the wrapper package
+to identify what information is retained.
+
+When a CUDS container is returned as a result of a ``get`` operation
+in a modelling engine the returned Container is a proxy to the
+information stored in the internal memory if the wrapper/modelling
+engine.
+
+CUDS containers
+---------------
+
+.. figure:: ./images/containers.png
+
+  **Figure. 2:** UML description of a CUDS Containers
+
+The CUDS containers, with the exception of the Lattice, are multi-type
+container (i.e. contain different types of objects). For each type the
+following operations are currently supported:
+
+* Add an item to the container
+* Get an item from the container based on the ``uid``
+* Remove an item from the container based on the ``uid``
```

Comment:
I have updated the uml diagram of the Mesh and added more info in the exceptions note